firebase-perf/src/main/java/com/google/firebase/perf/injection/modules/FirebasePerformanceModule.java

```diff
@@ -0,0 +1,94 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.perf.injection.modules;
+
+import androidx.annotation.NonNull;
+import com.google.android.datatransport.TransportFactory;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.inject.Provider;
+import com.google.firebase.installations.FirebaseInstallationsApi;
+import com.google.firebase.perf.config.ConfigResolver;
+import com.google.firebase.perf.internal.GaugeManager;
+import com.google.firebase.perf.internal.RemoteConfigManager;
+import com.google.firebase.remoteconfig.RemoteConfigComponent;
+import dagger.Module;
+import dagger.Provides;
+
+/**
+ * Provider for FirebasePerformance
+ *
+ * @hide
+ */
+@Module
+public class FirebasePerformanceModule {
+  private final FirebaseApp firebaseApp;
+  private final FirebaseInstallationsApi firebaseInstallations;
+  private final Provider<RemoteConfigComponent> remoteConfigComponentProvider;
+  private final Provider<TransportFactory> transportFactoryProvider;
+
+  //  private final RemoteConfigManager remoteConfigManager;
+  //  private final ConfigResolver configResolver;
+  //  private final GaugeManager gaugeManager;
+
+  public FirebasePerformanceModule(
+      @NonNull FirebaseApp firebaseApp,
+      @NonNull FirebaseInstallationsApi firebaseInstallations,
+      @NonNull Provider<RemoteConfigComponent> remoteConfigComponentProvider,
+      @NonNull Provider<TransportFactory> transportFactoryProvider) {
+    this.firebaseApp = firebaseApp;
+    this.firebaseInstallations = firebaseInstallations;
+    this.remoteConfigComponentProvider = remoteConfigComponentProvider;
+    this.transportFactoryProvider = transportFactoryProvider;
+
+    //    this.remoteConfigManager = RemoteConfigManager.getInstance();
+    //    this.configResolver = ConfigResolver.getInstance();
+    //    this.gaugeManager = GaugeManager.getInstance();
+  }
+
+  @Provides
+  FirebaseApp providesFirebaseApp() {
+    return firebaseApp;
+  }
+
+  @Provides
+  FirebaseInstallationsApi providesFirebaseInstallations() {
+    return firebaseInstallations;
+  }
+
+  @Provides
+  Provider<RemoteConfigComponent> providesRemoteConfigComponent() {
+    return remoteConfigComponentProvider;
+  }
+
+  @Provides
+  Provider<TransportFactory> providesTransportFactoryProvider() {
+    return transportFactoryProvider;
+  }
+
+  @Provides
+  RemoteConfigManager providesRemoteConfigManager() {
+    return RemoteConfigManager.getInstance();
+  }
+
+  @Provides
+  ConfigResolver providesConfigResolver() {
+    return ConfigResolver.getInstance();
+  }
+
+  @Provides
+  GaugeManager providesGaugeManager() {
+    return GaugeManager.getInstance();
+  }
```

Comment:
Should these APIs (for initializing `RemoteConfigManager`, `ConfigResolver`, `GaugeManager`) be part of `FirebasePerformanceModule`?

Also those objects are not initialized in Daggery way but I assume that's what the goal for this week.

Reply:
You are right. We will do this later this week in another PR.